design/system_tests.md

```diff
@@ -0,0 +1,42 @@
+# System Tests
+
+## Requirements
+1. Test the system in the same manner users will be using it. 
+   No ad-hoc initializations or mocking of any kind. 
+2. Cover the most common use-cases.
+3. Portability: The tests don't assume anything on the environment where the lakeFS & DB are running.
+   lakeFS address is the only **required** argument for running the tests.  
+
+## Non-Requirements
+1. Test every use-case of the system, including:
+    a. Every API/gateway endpoint.
+    b. Every application (e.g. Spark) usage pattern.
+2. Avoid repeating unit/integration tests scenarios.
+3. Stress testing / benchmarks. 
+
+## Solution
+### Test
+A test consists of one or more calls to the API/gateway and assertions on the actual output.
+Test is a standalone unit. It shouldn't rely on any other test to pass to start execution. 
+
+Later on we might consider adding checks on the DB or data lake layer(S3),
+by adding information of the underlying system components.
+
+#### Test-Cases
+TBD
+
+### Tests binary
+Tests will be built into a single binary and run sequentially or in parallel.
```

Comment:
This is a requirement? or good to have feature? what does it mean about test data?

Reply:
That's a feature that will allow us to add more tests without linear increase in the runtime.
Tests are isolated as I mentioned before, so matching test data should match that requirement too - nothing new here.

You can argue that we can start without it, but I think that it's a must have. Without it we'll soon reach long runtimes that will harm usability and adding more tests.